wasm/src/main/scala/ir2wasm/WasmExpressionBuilder.scala

```diff
@@ -1345,4 +1343,175 @@ private class WasmExpressionBuilder private (
 
     IRTypes.IntType
   }
+
+  private def genJSNew(tree: IRTrees.JSNew): IRTypes.Type = {
+    genTree(tree.ctor, IRTypes.AnyType)
+    genJSArgsArray(tree.args)
+    instrs += CALL(FuncIdx(WasmFunctionName.jsNew))
+    IRTypes.AnyType
+  }
+
+  private def genJSSelect(tree: IRTrees.JSSelect): IRTypes.Type = {
+    genTree(tree.qualifier, IRTypes.AnyType)
+    genTree(tree.item, IRTypes.AnyType)
+    instrs += CALL(FuncIdx(WasmFunctionName.jsSelect))
+    IRTypes.AnyType
+  }
+
+  private def genJSFunctionApply(tree: IRTrees.JSFunctionApply): IRTypes.Type = {
+    genTree(tree.fun, IRTypes.AnyType)
+    genJSArgsArray(tree.args)
+    instrs += CALL(FuncIdx(WasmFunctionName.jsFunctionApply))
+    IRTypes.AnyType
+  }
+
+  private def genJSMethodApply(tree: IRTrees.JSMethodApply): IRTypes.Type = {
+    genTree(tree.receiver, IRTypes.AnyType)
+    genTree(tree.method, IRTypes.AnyType)
+    genJSArgsArray(tree.args)
+    instrs += CALL(FuncIdx(WasmFunctionName.jsMethodApply))
+    IRTypes.AnyType
+  }
+
+  private def genLoadJSConstructor(tree: IRTrees.LoadJSConstructor): IRTypes.Type = {
+    val info = ctx.getClassInfo(tree.className)
+    val jsNativeLoadSpec = info.jsNativeLoadSpec.getOrElse {
+      throw new AssertionError(s"Found $tree for class without jsNativeLoadSpec at ${tree.pos}")
+    }
+    genLoadJSNativeLoadSpec(jsNativeLoadSpec)(tree.pos)
+  }
+
+  private def genLoadJSModule(tree: IRTrees.LoadJSModule): IRTypes.Type = {
+    val info = ctx.getClassInfo(tree.className)
+    val jsNativeLoadSpec = info.jsNativeLoadSpec.getOrElse {
+      throw new AssertionError(s"Found $tree for class without jsNativeLoadSpec at ${tree.pos}")
+    }
+    genLoadJSNativeLoadSpec(jsNativeLoadSpec)(tree.pos)
+  }
```

Comment:
[note]

```scala
js.Math.PI

// compiles to

JSSelect(LoadJSModule(ClassName<scala.scalajs.js.Math$>),StringLiteral(PI))
// where jsNativeLoadSpec is
Global(Math,List())
```

`genLoadJSNativeLoadSpec` will be a set of `JsGlobalRefGet` + `JsSelect` (for following the path)